yt_dlp/extractor/rokfin.py

```diff
@@ -160,6 +169,157 @@ def _get_comments(self, video_id):
+        self._clear_cookies()
+        self._access_mgmt_tokens = None
+        # No token revocation takes place during logout, as KEYCLOAK does not -- and has no plans to -- support individual token
+        # revocation on external party's request. See
+        # https://web.archive.org/web/20220215040021/https://keycloak.discourse.group/t/revoking-or-invalidating-an-authorization-token/1032
+
+    def _authentication_active(self):
+        current_time_utc = datetime.utcnow().timestamp()
+        SESSION_COOKIE_NAMES = {'KEYCLOAK_IDENTITY', 'KEYCLOAK_IDENTITY_LEGACY', 'KEYCLOAK_SESSION', 'KEYCLOAK_SESSION_LEGACY'}
+        return set([cookie.name for cookie in self._downloader.cookiejar if (cookie.name in SESSION_COOKIE_NAMES)
+                    and (cookie.name not in ('KEYCLOAK_SESSION', 'KEYCLOAK_SESSION_LEGACY')
+                         or cookie.expires > current_time_utc)]) == SESSION_COOKIE_NAMES
+
+    def _download_json_using_access_token(self, url_or_request, video_id, headers={}, query={}):
+        assert 'authorization' not in headers
+        headers = headers.copy()
+        authorization_hdr_val = try_get(self._access_mgmt_tokens, lambda tokens: tokens['token_type'] + ' ' + tokens['access_token'])
+        if authorization_hdr_val:
+            headers['authorization'] = authorization_hdr_val
+        json_string, urlh = self._download_webpage_handle(
+            url_or_request, video_id, note='Downloading JSON metadata' + (' [logged in]' if 'authorization' in headers else ''),
+            errnote='Unable to download JSON metadata' + (' [logged in]' if 'authorization' in headers else ''),
+            headers=headers, query=query, expected_status=401)  # 401=Unauthorized
+        if not authorization_hdr_val or urlh.code != 401 or self._access_mgmt_tokens.get('refresh_token') is None:
+            return self._parse_json(json_string, video_id)
+        del headers['authorization']
+        del self._access_mgmt_tokens['access_token']
+        del self._access_mgmt_tokens['token_type']
+        try:
+            del self._access_mgmt_tokens['expires_in']
+        except KeyError:
+            pass
+        self._access_mgmt_tokens.update(
+            self._download_json(
+                self._TOKEN_DISTRIBUTION_POINT_URL_STEP_6_7, video_id, note='Restoring lost authorization',
+                errnote='Failed to restore authorization', fatal=False,
+                data=urlencode_postdata({'grant_type': 'refresh_token', 'refresh_token': self._access_mgmt_tokens.get('refresh_token'), 'client_id': 'web'})) or {})
+        self.write_debug(f'Updated tokens: {self._access_mgmt_tokens.keys()}')
+        if {'access_token', 'expires_in', 'refresh_expires_in', 'refresh_token', 'token_type', 'id_token', 'not-before-policy', 'session_state', 'scope'} - self._access_mgmt_tokens.keys():
+            username, password = self._get_login_info()
+            self._logout()
+            self._perform_login(username, password)
```

Comment:
Have you observed re-logins helping with this site? If not, this is just overcomplicating things